crates/markdown_preview/src/markdown_renderer.rs

```diff
@@ -1,346 +1,322 @@
+fn render_markdown_heading(parsed: &ParsedMarkdownHeading, cx: &mut RenderContext) -> AnyElement {
+    let size = match parsed.level {
+        HeadingLevel::H1 => rems(2.),
+        HeadingLevel::H2 => rems(1.5),
+        HeadingLevel::H3 => rems(1.25),
+        HeadingLevel::H4 => rems(1.),
+        HeadingLevel::H5 => rems(0.875),
+        HeadingLevel::H6 => rems(0.85),
+    };
+
+    let color = match parsed.level {
+        HeadingLevel::H6 => cx.text_muted_color,
+        _ => cx.text_color,
+    };
+
+    let line_height = DefiniteLength::from(rems(1.25));
+
+    div()
+        .line_height(line_height)
+        .text_size(size)
+        .text_color(color)
+        .pt(rems(0.15))
+        .pb_1()
+        .child(render_markdown_text(&parsed.contents, cx))
+        .into_any()
 }
 
-impl<'a, I> Renderer<I>
-where
-    I: Iterator<Item = (Event<'a>, Range<usize>)>,
-{
-    fn new(
-        iter: I,
-        source_contents: String,
-        language_registry: &Arc<LanguageRegistry>,
-        theme: Arc<Theme>,
-    ) -> Self {
-        Self {
-            iter,
-            source_contents,
-            theme,
-            table: None,
-            finished: vec![],
-            language_registry: language_registry.clone(),
-            list_depth: 0,
-            block_quote_depth: 0,
-        }
-    }
+fn render_markdown_list(parsed: &ParsedMarkdownList, cx: &mut RenderContext) -> AnyElement {
+    use ParsedMarkdownListItemType::*;
 
-    fn run(mut self, cx: &WindowContext) -> Self {
-        while let Some((event, source_range)) = self.iter.next() {
-            match event {
-                Event::Start(tag) => {
-                    self.start_tag(tag);
-                }
-                Event::End(tag) => {
-                    self.end_tag(tag, source_range, cx);
-                }
-                Event::Rule => {
-                    let rule = div().w_full().h(px(2.)).bg(self.theme.colors().border);
-                    self.finished.push(div().mb_4().child(rule));
-                }
-                _ => {}
-            }
-        }
-        self
-    }
+    let mut items = vec![];
+    for item in &parsed.children {
+        let padding = rems((item.depth - 1) as f32 * 0.25);
 
-    fn start_tag(&mut self, tag: Tag<'a>) {
-        match tag {
-            Tag::List(_) => {
-                self.list_depth += 1;
-            }
-            Tag::BlockQuote => {
-                self.block_quote_depth += 1;
-            }
-            Tag::Table(column_alignments) => {
-                self.table = Some(MarkdownTable::new(
-                    self.theme.colors().border,
-                    column_alignments,
-                ));
-            }
-            _ => {}
-        }
-    }
+        let bullet = match item.item_type {
+            Ordered(order) => format!("{}.", order),
+            Unordered => "•".to_string(),
+            Task(checked) => if checked { "☑" } else { "☐" }.to_string(),
```

Comment:
might be not the best idea, but what about

`Task(checked) => if checked { "✅" } else { "◯" }.to_string(),`

probably too graphical, but pretty!

Reply:
We try to keep Zed from turning into some flashing, rainbow-like thing in its "base", plugin-less version, so not sure if introducing emojis right now is a good thing.

One note on the current style is that it looks somewhat small compared to the rest of my large font (Monocraft)
![image](https://github.com/zed-industries/zed/assets/2690773/106fa408-5223-4cc0-b9bf-ae2ff2221ee4)